src/Aspire.Dashboard/Model/ResourceStateViewModel.cs

```diff
@@ -42,7 +42,7 @@ private static (Icon icon, Color color) GetStateIcon(ResourceViewModel resource)
                 icon = new Icons.Filled.Size16.ErrorCircle();
                 color = Color.Error;
             }
-            else if (resource.IsFinishedState())
+            else if (resource.TryGetExitCode(out _) && (resource.IsFinishedState() || resource.IsExitedState()))
```

Comment:
What is TryGetExitCode here for?

Reply:
For example,

![image](https://github.com/user-attachments/assets/e6ddcaf2-eb9e-4273-b7b1-6b728314728b)